### ISSUE TITLE
Integration test failure: test_ansible_controller

### ISSUE THREAD

**Issue by TheRealFalcon** [canonical/cloud-init#4356]
`tests/integration_tests/modules/test_ansible.py::test_ansible_controller`
Looks to be a legit failure. Consistently happens on Lunar:
```
2023-08-15 23:11:13,710 - util.py[WARNING]: Running module ansible (<module 'cloudinit.config.cc_ansible' from '/usr/lib/python3/dist-packages/cloudinit/config/cc_ansible.py'>) failed
2023-08-15 23:11:13,710 - util.py[DEBUG]: Running module ansible (<module 'cloudinit.config.cc_ansible' from '/usr/lib/python3/dist-packages/cloudinit/config/cc_ansible.py'>) failed
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/cloudinit/config/modules.py", line 256, in _run_modules
    ran, _r = cc.run(
              ^^^^^^^
  File "/usr/lib/python3/dist-packages/cloudinit/cloud.py", line 67, in run
    return self._runners.run(name, functor, args, freq, clear_on_fail)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/cloudinit/helpers.py", line 172, in run
    results = functor(**args)
              ^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/cloudinit/config/cc_ansible.py", line 179, in handle
    ansible.install(package_name)
  File "/usr/lib/python3/dist-packages/cloudinit/config/cc_ansible.py", line 142, in install
    self.do_as([*cmd, "--upgrade", "pip"])
  File "/usr/lib/python3/dist-packages/cloudinit/config/cc_ansible.py", line 99, in do_as
    return self.distro.do_as(command, self.run_user, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/cloudinit/distros/__init__.py", line 1011, in do_as
    return subp.subp(
           ^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/cloudinit/subp.py", line 335, in subp
    raise ProcessExecutionError(
cloudinit.subp.ProcessExecutionError: Unexpected error while running command.
Command: ['su', '-', 'ansible', '-c', 'env PATH=$PATH /usr/bin/python3 -m pip install --user --upgrade pip']
Exit code: 1
Reason: -
Stdout: 
Stderr: error: externally-managed-environment
        
        × This environment is externally managed
        ╰─> To install Python packages system-wide, try apt install
            python3-xyz, where xyz is the package you are trying to
            install.
            
            If you wish to install a non-Debian-packaged Python package,
            create a virtual environment using python3 -m venv path/to/venv.
            Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
            sure you have python3-full installed.
            
            If you wish to install a non-Debian packaged Python application,
            it may be easiest to use pipx install xyz, which will manage a
            virtual environment for you. Make sure you have pipx installed.
            
            See /usr/share/doc/python3.11/README.venv for more information.
        
        note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
        hint: See PEP 668 for the detailed specification.

```

**Comment by dermotbradley** [comment]
I believe the same error would occur on Alpine also, the correct step would be to install the natively Alpine packaged "pip" via "apk add py3-pip".

**Comment by blackboxsw** [comment]
Adding priority label to fix for 23.3 upstream release

**Comment by holmanb** [comment]
This bug is a duplicate of https://github.com/canonical/cloud-init/issues/4244, closing.
